crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -185,56 +238,1250 @@ impl<S: States> NextState<S> {
+                      state_set: Option<Res<State<S::RawState>>>| {
+            if parent_changed.is_empty() {
+                return;
+            }
+            parent_changed.clear();
+
+            let new_state =
+                if let Some(state_set) = S::convert_to_usable_state(state_set.as_deref()) {
+                    T::exists(state_set)
+                } else {
+                    None
+                };
+
+            match new_state {
+                Some(value) => {
+                    if current_state.is_none() {
+                        internal_apply_state_transition(
+                            event,
+                            commands,
+                            current_state,
+                            Some(value),
+                        );
+                    }
+                }
+                None => {
+                    internal_apply_state_transition(event, commands, current_state, None);
+                }
+            };
         };
+
+        schedule
+            .add_systems(system.in_set(ApplyStateTransition::<T>::apply()))
+            .add_systems(
+                apply_state_transition::<T>.in_set(StateTransitionSteps::ManualTransitions),
+            )
+            .add_systems(
+                should_run_transition::<T, OnEnter<T>>
+                    .pipe(run_enter::<T>)
+                    .in_set(StateTransitionSteps::EnterSchedules),
+            )
+            .add_systems(
+                should_run_transition::<T, OnExit<T>>
+                    .pipe(run_exit::<T>)
+                    .in_set(StateTransitionSteps::ExitSchedules),
+            )
+            .add_systems(
+                should_run_transition::<T, OnTransition<T>>
+                    .pipe(run_transition::<T>)
+                    .in_set(StateTransitionSteps::TransitionSchedules),
+            )
+            .configure_sets(
+                ApplyStateTransition::<T>::apply()
+                    .in_set(StateTransitionSteps::DependentTransitions)
+                    .after(ApplyStateTransition::<S::RawState>::apply()),
+            );
+    }
+}
+/// Trait defining a state that is automatically derived from other [`States`].
+///
+/// A Sub State is a state that exists only when the source state meet certain conditions,
+/// but unlike [`ComputedStates`] - while they exist they can be manually modified.
+///
+/// The [`StateSet`] is passed into the `exist` method whenever one of them changes, and the
+/// result is used to handle it's existence. If the result is `Some(Self)`, and the state doesn't exist,
+/// the state is set to the provided value. If it is `None`, the state is removed. Otherwise - the computation
+/// is not used to impact the state's value at all.
+///
+/// The default approach to creating [`SubStates`] is using the derive macro, and defining a single source state
+/// and value to determine it's existence. Note that this approach requires implementing [`Default`] as well.
```

Comment:
```suggestion
/// and value to determine it's existence.
```
I don't think this is a helpful detail to mention here.